azure_img_utils/cli/cli_utils.py

```diff
@@ -0,0 +1,192 @@
+# -*- coding: utf-8 -*-
+
+"""Azure image utils cli module utils."""
+
+# Copyright (c) 2022 SUSE LLC
+#
+# This file is part of azure_img_utils. azure_img_utils provides an
+# api and command line utilities for handling images in the Azure Cloud.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import click
+import logging
+import os
+import sys
+import yaml
+
+from collections import namedtuple, ChainMap
+
+
+default_config_dir = os.path.expanduser('~/.config/azure_img_utils/')
+default_profile = 'default'
+
+config_defaults = {
+    'config_dir': default_config_dir,
+    'profile': default_profile,
+    'log_level': logging.INFO,
+    'no_color': False,
+    'sas_token': None,
+    'tenant_id': None,
+    'client_id': None,
+    'client_secret': None,
+    'subscription_id': None,
+    'active_directory_url': 'https://login.microsoftonline.com',
+    'resource_manager_url': 'https://management.azure.com/',
+    'active_directory_graph_res_url': 'https://graph.windows.net/',
+    'sql_management_url': 'https://gallery.azure.com/',
+    'gallery_url': 'https://management.core.windows.net:8443/',
+    'management_url': 'https://management.core.windows.net/',
+    'credentials_file': None,
+    'resource_group': None,
```

Comment:
Last piece for config, we'll want to make sure the available configuration options match up with defaults if there can be a default. I.e. we can drop any values from default that aren't configuration options.

Reply:
Yep, I missed those